code/obj/item/gun/kinetic.dm

```diff
@@ -2144,6 +2144,212 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 		else
 			boutput(user, SPAN_ALERT("You can't fire \the [src] when it is open!"))
 
+#define ONE_BARREL 1
+#define TWO_BARRELS 2
+#define ALL_BARRELS 4
+#define MAX_USES 9
+
+/obj/item/gun/kinetic/sawnoff/quadbarrel //for salvagers
+
+	name = "\improper Four Letter Word"
+	desc = "For when you REALLY need to get the point across."
+	icon = 'icons/obj/large/64x32.dmi'
+	icon_state = "quadb"
+	item_state = "quadbarrel" //custom inhands, though.
+
+	camera_recoil_sway_min = 10 //VIOLENCE!
+	recoil_strength = 15
+	recoil_max = 60
+
+	var/guaranteed_uses = MAX_USES	// allows for just over two volleys before it starts breaking
+	var/firemode = ONE_BARREL
+	var/priorammo
+
+	force = MELEE_DMG_RIFLE
+	max_ammo_capacity = 4
+	two_handed = TRUE
+
+	New()
+		..()
+		ammo = new/obj/item/ammo/bullets/pipeshot/glass
+		set_current_projectile(new/datum/projectile/special/spreader/buckshot_burst/glass)
```

Comment:
Seems like this should start out empty or with a non-player created ammo. But that could just be a no ammo subtype.